lib/TWCManager/TWCMaster.py

```diff
@@ -626,6 +657,18 @@ def loadSettings(self):
         carapi.setCarApiRefreshToken(self.settings.get("carApiRefreshToken", ""))
         carapi.setCarApiTokenExpireTime(self.settings.get("carApiTokenExpireTime", ""))
 
+        # If particular details are missing from the Settings dict, create them
+        if not self.settings.get("VehicleGroups", None):
+            self.settings["VehicleGroups"] = {}
+        if not self.settings["VehicleGroups"].get("Allow Charging", None):
+            self.settings["VehicleGroups"]["Allow Charging"] = {
+                "Members": []
+            }
+        if not self.settings["VehicleGroups"].get("Deny Charging", None):
+            self.settings["VehicleGroups"]["Deny Charging"] = {
+                "Members": []
+            }
```

Comment:
Unless you plan to add additional properties on these later, the Members member seems superfluous.  Any reason Allow/Deny aren't simply arrays themselves?  When you access the lists, you can `.get(..., [])` to ensure you get a valid array back if they're missing.

Reply:
The reason for this is that it had occurred to me that we might want additional properties for groups in the future, for example - we could have a group parameter where we could add vehicles and specify a maximum rate for reduced charge rate for some vehicles. Not sure if/how fast this would ever be implemented or needed but there's some very creative use cases around